### ISSUE TITLE
[Feature request] Rate limit handling

### ISSUE THREAD

**Issue by tianalemesle** [zygos/mailerlite-api-v2-node#23]
- Automatically retry requests returning `429` using `setTimeout` based on the `X-RateLimit-Retry-After` response header
- Automatically delay further requests on same endpoints using the same header

Thanks

**Comment by zygos** [comment]
Interesting idea, will look into it.

**Comment by balajikummari** [comment]
Yes please, Will be really handy.

**Comment by tianalemesle** [comment]
Yes, because :
> 60 requests on individual endpoints can be made per minute

That's too easy to reach.
Thanks

**Comment by Rush** [comment]
> 60 requests on individual endpoints can be made per minute

this is not true. The bulk endpoint is limited differently and it's counting each individual bulk request into the global limit :o
